app/scripts/modules/core/src/pipeline/config/triggers/artifacts/gcs/defaultGcs.artifact.ts

```diff
@@ -0,0 +1,58 @@
+import { module } from 'angular';
+
+import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
+import { IArtifact } from 'core/domain/IArtifact';
+import { PipelineConfigProvider } from 'core/pipeline';
+import { isNullOrUndefined } from 'util';
+
+export const DEFAULT_GCS_ARTIFACT = 'spinnaker.core.pipeline.trigger.artifact.defaultGcs';
+module(DEFAULT_GCS_ARTIFACT, [
+  PIPELINE_CONFIG_PROVIDER,
+]).config((pipelineConfigProvider: PipelineConfigProvider) => {
+  pipelineConfigProvider.registerArtifactKind({
+    label: 'GCS',
+    description: 'A GCS object.',
+    key: 'default.gcs',
+    isDefault: true,
+    isMatch: false,
+    controller(artifact: IArtifact) {
+      'ngInject';
+      this.artifact = artifact;
+      this.artifact.type = 'gcs/object';
+
+      this.onNameChange = () => {
```

Comment:
`onReferenceChange` since ng-model is artifact.reference & for consistency with defaultDocker.artifact.ts?

Reply:
yup!

Reply:
Oh wait, you caught my typo